slackapi/deno-slack-hooks
# chore: update esbuild dependencies for bundling

### Summary

This PR update the `esbuild` related dependency and improves support for bundling npm packages.

It also gets ride of the `deps.ts` file. This prevents every hook from importing all the project dependencies when executing.

### Testing

1. Create a new app
2. Use the following `slack.json`
```json
{
  "hooks": {
    "get-hooks": "deno run -q --allow-read --allow-net path/to/deno-slack-hooks/src/mod.ts",
    "get-manifest": "deno run -q --config=deno.jsonc --allow-read --allow-net --allow-env --allow-sys=osRelease path/to/deno-slack-hooks/src/get_manifest.ts",
    "build": "deno run -q --config=deno.jsonc --allow-read --allow-write --allow-net --allow-run --allow-env --allow-sys=osRelease path/to/deno-slack-hooks/src/build.ts",
    "get-trigger": "deno run -q --config=deno.jsonc --allow-read --allow-net --allow-env path/to/deno-slack-hooks/src/get_trigger.ts",
    "check-update": "deno run -q --config=deno.jsonc --allow-read --allow-net path/to/deno-slack-hooks/src/check_update.ts",
    "install-update": "deno run -q --config=deno.jsonc --allow-run --allow-read --allow-write --allow-net path/to/deno-slack-hooks/src/install_update.ts",
    "doctor": "deno run -q --config=deno.jsonc --allow-read --allow-net path/to/deno-slack-hooks/src/doctor.ts"
  }
}

```
4. Import the `import { WebClient } from "npm:@slack/web-api";`
5. The app should bundle properly, but it will fail at runtime since `@slack/web-api` requires `--allow-sys=osRelease` to work.

### Requirements (place an `x` in each `[ ]`)
